### ISSUE TITLE
Problems with clicking on SwtichRow's GtkSwitch handle

### ISSUE THREAD

**Issue by E-D-W-I-N** [kaii-lb/overskride#17]
Hi, first of all thank you for your hard work, this is a really cool tool :)

Today I've tried overskride for the first time and everything works great except one really weird behavior - If I click on switch handle  inside SwitchRow nothing happens. Literally every switch (powered, device connection etc.) does nothing, but if I click not on switch but on whole SwitchRow's body my click finally works.

According to libadwaita docs this feels like expected behavior (https://gnome.pages.gitlab.gnome.org/libadwaita/doc/main/class.SwitchRow.html)

> The AdwSwitchRow widget contains a [GtkSwitch](https://docs.gtk.org/gtk4/class.Switch.html) that allows the user to select between two states: “on” or “off”. When activated, the row will invert its active state.
> 
> The user can control the switch by activating the row or by dragging on the switch handle. 

Nothing about clicks on switch handle.

Idk maybe this problem is specific to my device and on your side everything works well, let me know if so pls.

By the way, I've also have Bottles (gaming wine tool) installed, it also uses libadwaita but clicks on switch handle in Bottles works ok. I've checked Bottles source code and found that they handle switch handle (haha, lol) separately inside ActionRow - https://github.com/bottlesdevs/Bottles/blob/f9c19ae7cd2c830ee085aad30c8076207331a7e0/bottles/frontend/ui/preferences.blp#L17

I guess that's why both clicking on row body and switch handle works just right.

Anyways this shit is minor af, so, you shouldn't probably waste your time on stuff like that now, but at least we can check if problem is only on my side or not

Thanks!

**Comment by kaii-lb** [comment]
It is expected behavior, and i was too lazy to handle it but since it seems tk be an issue i'll try to fix it sometimes Wednesday lr this weekend. cant promise anything cuz skl

**Comment by E-D-W-I-N** [comment]
yeah it's pretty minor, I guess you could just write a little note in readme and leave it as it is

**Comment by Fxxxlei** [comment]
This is only a minor problem if you know about it.
I've been trying to find out what issue I was having for over half an hour, always intuitively clicking on the toggle itself.

For some reason the toggle switches, giving visual feedback, but no action is taken.

**Comment by kaii-lb** [comment]
yeah it seems to be a bug with how gnome makes their switch rows. imma need to redesign the row itself to do that but currently i have midyear exams, so i dont have time

**Comment by sleklere** [comment]
Fix is up at #59. Short version: `GtkSwitch` claims the click itself, so the
row never activates and `activated` never fires — `notify::active` catches both
click targets. Longer writeup in #25.
